### ISSUE TITLE
TransferManager is unable to copy S3 objects between regions

### ISSUE THREAD

**Issue by katjes733** [aws/aws-sdk-java#2076]
Opening new issue because the following issue is not resolved with the current version (1.11.608): https://github.com/aws/aws-sdk-java/issues/1676

TransferManager is unable to copy S3 objects between regions (same account) using [this](https://docs.aws.amazon.com/zh_cn/AWSJavaSDK/latest/javadoc/com/amazonaws/services/s3/transfer/TransferManager.html#copy-com.amazonaws.services.s3.model.CopyObjectRequest-com.amazonaws.services.s3.AmazonS3-com.amazonaws.services.s3.transfer.internal.TransferStateChangeListener-) method
(`TransferManager.copy(CopyObjectRequest copyObjectRequest, AmazonS3 srcS3, TransferStateChangeListener stateChangeListener)`)
, which, according to the documentation is supposed to be the one to use for cross region operations.
In my example, I want to copy from region eu-central-1 to ap-southeast-1. 
1. If I set the AmazonS3 client to eu-central-1, I get the following error: 
`com.amazonaws.services.s3.model.AmazonS3Exception: The bucket is in this region: ap-southeast-1. Please use this region to retry the request`
2. If I set the AmazonS3 client to ap-southeast-1, I get the following error:
`com.amazonaws.services.s3.model.AmazonS3Exception: The bucket is in this region: eu-central-1. Please use this region to retry the request`

The documentation for parameter srcS3 even states: 
> srcS3 - An AmazonS3 client constructed for the region in which the source object's bucket is located

Therefore I actually expected that option 1 worked.

Interestingly, if I perform my copy operation using [this](https://docs.aws.amazon.com/AWSJavaSDK/latest/javadoc/com/amazonaws/services/s3/AmazonS3.html#copyObject-com.amazonaws.services.s3.model.CopyObjectRequest-) method
(`AmazonS3.copyObject(CopyObjectRequest copyObjectRequest)`) the following happens:
1. If I set the AmazonS3 client to eu-central-1, I get the following error: 
`com.amazonaws.services.s3.model.AmazonS3Exception: The bucket is in this region: ap-southeast-1. Please use this region to retry the request`
2. If I set the AmazonS3 client to ap-southeast-1, the request completes successfully.

To summarize, the `Transfer.copy` between regions is not working and it is a bit counter intuitive to have to use the target bucket region when using `Amazon.copyObject`

**Comment by katjes733** [comment]
I missed this:
```java
    public static void main(String[] args) throws InterruptedException {
        AmazonS3 sourceS3 = AmazonS3ClientBuilder.standard()
                                                 .withRegion(SOURCE_REGION)
                                                 .build();
        AmazonS3 destintationS3 = AmazonS3ClientBuilder.standard()
                                                       .withRegion(DESTINATION_REGION)
                                                       .build();

        TransferManager transferManager = TransferManagerBuilder.standard()
                                                                .withS3Client(destintationS3)
                                                                .build();

        Copy copy = transferManager.copy(new CopyObjectRequest(SOURCE_BUCKET_NAME, SOURCE_KEY,
                                                               DESTINATION_BUCKET_NAME, DESTINATION_KEY),
                                         sourceS3, null);


        copy.waitForCopyResult();
        transferManager.shutdownNow();
        sourceS3.shutdown();
        destintationS3.shutdown();
    }
```

_Originally posted by @shorea in https://github.com/aws/aws-sdk-java/issues/1676#issuecomment-405440525_


Now my code works, but it was really not clear for me from the documentation, that I needed the TransferManager to point to the destination AmazonS3 client.

**Comment by freshmanken** [comment]
Does the same support in S3TransferManager V2? Didn't see a way to do the similar

**Comment by kedbirhan** [comment]
i have been trying to do the same thing in V2, any help is appreciated?
